### PR TITLE
100: Implement Block->transactionsInvolving field.

### PR DESCRIPTION
**Pull request by akhila-raju**

### CODE REVIEW

File: src/schema/core.graphql [hunk 1]
```diff
@@ -86,7 +86,7 @@ type Block {
   Gets all transactions from this block as long as they involve any of the addresses specified.
   If a filter is passed, only the transactions matching the filter will be returned.
   """
-  transactionsInvolving(participants: [String]!, filter: TransactionFilter): [Transaction]
+  transactionsInvolving(participants: [Address]!, filter: TransactionFilter): [Transaction]
```
Comment (raulk) [inline src/schema/core.graphql]: Good catch here!

File: src/resolvers/block.ts [hunk 2]
```diff
@@ -89,6 +89,33 @@ const blocksRange: IFieldResolver<any, any> = async (obj, { numberRange, hashRan
 const transactions = (obj, args) =>
   obj.transactions.map(tx => ({ ...tx, from: { address: tx.from }, to: { address: tx.to }, inputData: tx.input }));
 
+interface ITransactionsInvolvingArgs {
+  participants?: [string];
+}
+  
+const transactionsInvolving: IFieldResolver<any, any> = async (obj, { participants }: ITransactionsInvolvingArgs) => {
```
Comment (raulk) [inline src/resolvers/block.ts]: I see that this makes a JSON-RPC request for each transaction in the block. Given that `obj` is already a block that came from `web3.eth.getBlock(..., true)`, and hence it contains all transaction data (`obj.transactions`), would it not be more efficient to traverse the transactions we already have, and select only the ones that match the criteria? You should be able to achieve with `obj.transactions.filter(tx => ...)`.

File: src/providers/web3.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
 import Web3 = require('web3');
 
 const http = new Web3.providers.HttpProvider(`https://mainnet.infura.io/${process.env.INFURA_ID}`);
-export const web3 = new Web3(http);
+export const web3 = new Web3(http);
```
Comment (raulk) [inline src/providers/web3.ts]: Looks like the linting Git hook is not being called?

File: src/resolvers/block.ts (hunk 2 above)
Comment (akhila-raju) [reply src/resolvers/block.ts]: SGTM

File: src/providers/web3.ts (hunk 3 above)
Comment (akhila-raju) [reply src/providers/web3.ts]: Argh :|

**Pull request merged**